File: app/src/main/java/com/mmjang/ankihelper/data/ai/AITranslatorConfig.java
package com.mmjang.ankihelper.data.ai;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(
    tableName = "aitranslatorconfig",
    foreignKeys = @ForeignKey(
        entity = LLMConfig.class,
        parentColumns = "id",
        childColumns = "llmId",
        onDelete = ForeignKey.CASCADE
    ),
    indices = @Index("llmId")
)
public class AITranslatorConfig {
    @PrimaryKey(autoGenerate = true)
    private long id;
    @ColumnInfo(name = "translatorName")
    private String translatorName;

    @ColumnInfo(name = "llmId")
    private long llmId; // Foreign key to LLMConfig

    private String prompt;

    @ColumnInfo(name = "isDefault")
    private boolean isDefault;

    @ColumnInfo(name = "sourceLanguage")
    private String sourceLanguage;

    @ColumnInfo(name = "targetLanguage")
    private String targetLanguage;
    
    // Getters and setters
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTranslatorName() {
        return translatorName;
    }

    public void setTranslatorName(String translatorName) {
        this.translatorName = translatorName;
    }

    public long getLlmId() {
        return llmId;
    }

    public void setLlmId(long llmId) {
        this.llmId = llmId;
    }

    public String getPrompt() {
        return prompt;
    }

    public void setPrompt(String prompt) {
        this.prompt = prompt;
    }

    public boolean isDefault() {
        return isDefault;
    }

    public void setDefault(boolean aDefault) {
        isDefault = aDefault;
    }

    public String getSourceLanguage() {
        return sourceLanguage;
    }

    public void setSourceLanguage(String sourceLanguage) {
        this.sourceLanguage = sourceLanguage;
    }

    public String getTargetLanguage() {
        return targetLanguage;
    }

    public void setTargetLanguage(String targetLanguage) {
        this.targetLanguage = targetLanguage;
    }
}

File: app/src/main/java/com/lmyby/ankihelper/data/ai/AITranslatorConfig.kt
package com.lmyby.ankihelper.data.ai

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

/**
 * Converted to Kotlin as part of Phase 2 data model migration
 */
@Entity(
    tableName = "aitranslatorconfig",
    foreignKeys = [ForeignKey(
        entity = LLMConfig::class,
        parentColumns = ["id"],
        childColumns = ["llmId"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index("llmId")]
)
data class AITranslatorConfig(
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0,

    @ColumnInfo(name = "translatorName")
    var translatorName: String? = null,

    @ColumnInfo(name = "llmId")
    var llmId: Long = 0, // Foreign key to LLMConfig

    var prompt: String? = null,

    @ColumnInfo(name = "isDefault")
    var isDefault: Boolean = false,

    @ColumnInfo(name = "sourceLanguage")
    var sourceLanguage: String? = null,

    @ColumnInfo(name = "targetLanguage")
    var targetLanguage: String? = null
)
